agent/eni/iphelperwrapper/iphelper_windows.go

```diff
@@ -0,0 +1,168 @@
+// +build windows
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package iphelperwrapper
+
+import (
+	"C"
+	"math"
+	"syscall"
+	"time"
+	"unsafe"
+
+	log "github.com/cihub/seelog"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+const (
+	// After registering a callback with NotifyIPInterfaceChange API, we get an initial notification to confirm that the callback is registered
+	// All other notifications sent on the channel would be positive except this one
+	InitialNotification = math.MinInt16
+)
+
+var (
+	// notificationChannel is used for communication between the notification callback from Windows APIs and the consumer
+	notificationChannel chan int
+
+	// initialNotificationTimeout is the timeout for the initial notification.
+	// If notification is not received within this timeout then we will assume error
+	// We have made this as a var so as to override this value in tests
+	initialNotificationTimeout = 4 * time.Second
+)
+
+// InterfaceMonitor is used for monitoring interface additions
+// InterfaceMonitor can be mocked while unit testing other packages
+type InterfaceMonitor interface {
+	Start(notification chan int) error
+	Close() error
+}
+
+type eniMonitor struct {
+	notificationHandle          windows.Handle
+	funcNotifyIPInterfaceChange func(a ...uintptr) (r1 uintptr, r2 uintptr, lastErr error)
+	funcCancelMibChangeNotify2  func(a ...uintptr) (r1 uintptr, r2 uintptr, lastErr error)
+}
+
+// NewMonitor creates a new monitor to implement the interface
+func NewMonitor(params ...func(a ...uintptr) (r1 uintptr, r2 uintptr, lastErr error)) InterfaceMonitor {
```

Comment:
NewMonitor returns an Interface and hence we cannot use it to inject mocks for funcNotifyIPInterfaceChange and funcCancelMibChangeNotify2.
Therefore, we take variable inputs for our function to inject the mocks. If no param is provided then we take the default functions.